### PR TITLE
Add Serverless stats path

### PR DESCRIPTION
**Pull request by JasonEtco**
This PR adds support for HTTP GET routes in Serverless.

Currently, there are two major errors:

1. Hitting any path results in the following warning being logged to console, and adds another warning every time the route is requested (resulting in a mess of duplicate warnings)

> WARNING: Attempting to listen on socket /tmp/server19.sock, but it is already in use. This is likely as a result of a previous invocation error or timeout. Check the logs for the invocation(s) immediately prior to this for root cause, and consider increasing the timeout and/or cpu/memory allocation if this is purely as a result of a timeout. aws-serverless-express will restart the Node.js server listening on a new port and continue with this request.

2. When navigating to `/probot`, the following error is returned:

> Error: Failed to lookup view "probot.ejs" in views directory "/views"

I'm sure that I had fixed it in `probot/probot`. This error does not occur in my own DigitalOcean Droplet, so I'm confused.

cc @tcbyrd - The paths "work" so thats nice.

### CODE REVIEW

File: handler.js [hunk 1]
```diff
@@ -16,33 +17,39 @@ const probot = createProbot({
 })
 
 // Load Probot application
-probot.load(todo)
+probot.setup([todo])
```
Comment (tcbyrd) [inline handler.js]: This is what I was missing and what we should probably document in the lambda handler. Is the difference that `probot.load` overrides the default plugins, while `probot.setup` allows you to add other plugins?

Comment (JasonEtco) [reply handler.js]: `probot.setup()` calls `probot.load()` on an array of default plugins *and* whatever other plugins you pass it

Comment (tcbyrd) [reply handler.js]: Gotcha. So calling `probot.load`, it never even looks for the default plugins.

**Comment by tcbyrd** [comment]
> Error: Failed to lookup view "probot.ejs" in views directory "/views"

Basically an error caused by bundling. The way it's transpiling the `probot` directory, the relative links to other directories is messed up. My lazy hack to transpile the probot dependency breaks it. Still looking for the best way to make webpack understand what I want it to do.

**Comment by JasonEtco** [comment]
Ooooh @tcbyrd you're totally right. I'll put some more thinking towards a better webpack solution then (again), see if I can figure out any better way of doing it.

**Comment by tcbyrd** [comment]
This may help:
https://webpack.js.org/guides/dependency-management/#require-context
> It allows you to pass in a directory to search, a flag indicating whether subdirectories should be searched too, and a regular expression to match files against.

**Comment by tcbyrd** [comment]
Also just want to point out that `/stats` will need to work differently on Lambda. Currently the interval is one an hour. Max execution time for Lambda is 300 seconds. Is there any reason we couldn't just have the endpoint get the data when it's invoked? Alternatively, we can call a [schedule event](https://serverless.com/framework/docs/providers/aws/events/schedule/), but the question then becomes where do we store that data?

**Comment by JasonEtco** [comment]
Ah that's a good point. The question of whether or not the refresh is necessary is probably best answered by @bkeepers; I see the use of it (to poll for popular installations), so I'm wondering if we can just pass a config option somehow to disable it.

Alternatively, we can copy-pasta the `/stats` endpoint logic into a separate route, but that feels hacky.

**Comment by tcbyrd** [comment]
@JasonEtco I talked to @bkeepers about the stats endpoint earlier. Basically because of rate limiting and the way we have to build this data, a larger installation could take a while to build up the data and would have to make multiple calls. I'm thinking the "right" thing to get it working for now is to create a scheduled function that stores the results in an s3 bucket. Then the stats endpoint on lambda would just respond with that object. Thoughts? Effectively, we need a way to store the results so that it persists beyond the execution of the function.

**Comment by JasonEtco** [comment]
> create a scheduled function that stores the results in an s3 bucket

That's an interesting solution. Can `serverless` also spin up an S3 bucket? Ideally, we'd want the cognitive overhead for this task to be as minimal as possible.

**Comment by tcbyrd** [comment]
>Can serverless also spin up an S3 bucket?

Yes. In fact, it already does. The zip package and CloudFormation template it generates are stored in a bucket with the name of your serverless service, so the scheduled function could just use that existing resources. I'm going to play around with this idea today.

**Comment by JasonEtco** [comment]
We're working on some overhauls to how we do serverless (using Typescript to compile Probot for us), so I'm going to close this until that's happened, since I'm sure much will change. 🎉

**Pull request closed without merging**